dags/map_reproducibility/utils/common_utils.py

```diff
@@ -19,55 +19,68 @@
 from google.cloud import storage
 import yaml
 
-
-def set_variables_cmds():
-  set_variables = (
-      "export PROJECT=supercomputer-testing",
-      "export CLUSTER=a3plus-benchmark",
-      "export CLUSTER_REGION=australia-southeast1",
-      "NOW=$(date +%s)",
-      "export BUCKET_NAME=regression-testing-xlml",
-  )
-  return set_variables
-
-
-def configure_project_and_cluster():
-  set_project_command = (
-      "gcloud config set project $PROJECT",
-      "sudo chown -R airflow:airflow /home/airflow/composer_kube_config",
-      "gcloud container clusters get-credentials "
-      "$CLUSTER --region $CLUSTER_REGION",
-  )
-  return set_project_command
+PROJECT = "supercomputer-testing"
+BUCKET_NAME = "regression-testing-xlml"
 
 
 # This is required to get auth to access
 def git_cookie_authdaemon():
   auth_cmds = (
       "git clone https://gerrit.googlesource.com/gcompute-tools",
       "echo 'trying to run git-cookie-authdaemon'",
-      # Check if the daemon is already running
-      "if pgrep -f git-cookie-authdaemon; then "
-      "  echo 'git-cookie-authdaemon is already running'; "
-      "else "
-      "  ./gcompute-tools/git-cookie-authdaemon || echo 'Error running git-cookie-authdaemon'; "  # Run if not running
-      "fi",
+      "  ./gcompute-tools/git-cookie-authdaemon || echo 'Error running git-cookie-authdaemon'",
   )
   return auth_cmds
 
 
-def clone_gob():
+def clone_recipes_gob():
   gob_clone_cmds = (
       "echo 'trying to clone GoB repo from outside'",
       "git clone https://ai-hypercomputer-benchmarks.googlesource.com/"
       "reproducible-benchmark-recipes",
-      "cd reproducible-benchmark-recipes/projects",
-      "cd gpu-recipes",
-      "pwd",
   )
   return gob_clone_cmds
 
 
+def get_aotc_repo():
+  gob_clone_cmds = (
+      "echo 'trying to clone GoB aotc repo'",
+      "git clone https://cmcs-perf-tooling-internal.googlesource.com/"
+      "benchmark-automation",
+  )
+  return gob_clone_cmds
+
+
+def configure_project_and_cluster(cluster: str, cluster_region: str):
+  set_project_command = (
+      f"gcloud config set project {PROJECT}",
+      "sudo chown -R airflow:airflow /home/airflow/composer_kube_config",
+      "gcloud container clusters get-credentials "
+      f"{cluster} --region {cluster_region}",
+  )
+  return set_project_command
+
+
+def get_gpu_recipe_cmd(hypercomputer, model_id, framework, recipe_repo_root):
+  gpu_recipe_cmd = (
+      "cd reproducible-benchmark-recipes/projects/gpu-recipes",
+      "export RECIPE_ROOT="
+      f"{recipe_repo_root}/training/{hypercomputer}/{model_id}/{framework}-pretraining-gke",
+      "cd $RECIPE_ROOT",
+  )
+  return gpu_recipe_cmd
+
+
+def get_pre_workload_cmds(model_id, framework, image_version):
+  prepare_workload_cmds = (
+      "NOW=$(date +%s)",
+      f"export JOB_NAME={model_id}-xlml-$NOW-{framework}",
+      "DOCKER_IMAGE=us-central1-docker.pkg.dev/"
+      f"supercomputer-testing/gunjanjalori/{framework}_test/{image_version}",
```

Comment:
I think the image should be specified by each recipe, because we are going to switch to AotC recipe for A3U, then the image will be the AotC image instead of the general nemo image.

Reply:
or you can pass that as variable defined in each recipe